go-fuzz-build/main.go

```diff
@@ -58,10 +58,34 @@ func makeTags() string {
 // that clients can then modify and use for calls to go/packages.
 func basePackagesConfig() *packages.Config {
 	cfg := new(packages.Config)
-	cfg.Env = append(os.Environ(), "GO111MODULE=off")
+
+	// Preliminary modules support: note that we do not set 
+	// GO111MODULE here in order to respect any GO111MODULE setting by the user
+	// as we are finding dependencies. Note, however, that we are
+	// still setting up a GOPATH to build, so we later will force 
+	// GO111MODULE to be off when building so that we are in GOPATH mode.
+	// If the user has not set GO111MODULE, the meaning here is
+	// left up to cmd/go (defaulting to 'auto' in Go 1.11-1.13,
+	// but likely defaulting to 'on' at some point during Go 1.14
+	// development cycle).
+	// Also note that we are leaving the overall cfg structure
+	// in place to support future experimentation, etc.
+	cfg.Env = os.Environ()
 	return cfg
 }
 
+// checkModVendor reports if the GOFLAGS env variable
+// contains -mod=vendor, which enables vendoring for modules.
+func checkModVendor() bool {
+	val := os.Getenv("GOFLAGS")
```

Comment:
I don't think this is correct; what if it was set via `go env -w GOFLAGS=-mod=vendor`?

I think the only portable way is exec'ing `go env GOFLAGS`.

Reply:
> I think the only portable way is exec'ing `go env GOFLAGS`.

The code here I wrote roughly two months ago was written before Go 1.13 was released, and of course Go 1.12 does not support `go env -w`, but you are right that that check does not look for whether or not someone used the new Go 1.13 `go env -w` behavior.

That said, your suggested alternative does not handle tip / Go 1.14 behavior.

I could do something like inspect the output of `go list` to see if a vendor directory is in use, or another solution. It's solvable...

...but rather than investing more time in chasing a moving target, if people don't want a friendlier error that currently targets what is (arguably) the more common case currently of how someone could try to enable vendoring, then I would prefer to drop this check.

If we remove the current check, one of the current tests show what the error would be in that case if someone tries to use vendoring, which states that `go-fuzz-dep` cannot be found  `vendor` directory:

`open ...\foo\vendor\github.com\dvyukov\go-fuzz\go-fuzz-dep: The system cannot find the path specified`

Note if we were to remove the "friendlier erorr" check, then the "less friendly" error still says `vendor` in the error message, so perhaps that is enough of a hint.

Rather than chasing a moving target for an error message that will ultimately go away, I think a better solution would be to apply that energy against someone implementing #252, which means this error in question would likely go away. I likely won't get to #252 myself in the near term.

Reply:
Sounds good to me. When I made this comment, automatic vendoring in module mode wasn't merged yet :) It certainly is a moving target.